oak_functions/loader/src/server.rs

```diff
@@ -145,13 +144,25 @@ pub trait OakApiNativeExtension {
     /// Performs any cleanup or terminating behavior necessary before destroying the WasmState.
     fn terminate(&mut self) -> anyhow::Result<()>;
 }
+
 pub trait ExtensionFactory {
     fn create(&self) -> anyhow::Result<BoxedExtension>;
 }
 
-pub type BoxedExtension = Box<dyn OakApiNativeExtension + Send + Sync>;
+pub enum Extension {
+    Native(Box<dyn OakApiNativeExtension + Send + Sync>),
+    Uwabi(Box<dyn UwabiExtension + Send + Sync>),
```

Comment:
I think it would be useful to have this explanation as a doc comment on `BoxedExtension`.

I am not sure about keeping two copies of the same logic. Does it mean that we'd have to maintain both and keep them in sync? Anyways, I think we can decide about this later :)

Reply:
Added a comment on BoxedExtension.

I see your concern about synching. The main difference between the two is whether the extensions is invoked through a dedicated ABI function or through a message on the channel. The actual logic behind the extension should remain the same.

I agree we'll see later if we think that is a good idea :)

Reply:
If the actual logic is to be implemented only in one place, then I think it is good to have both :)